packages/react-native-sdk/src/components/Profile.tsx

```diff
@@ -0,0 +1,434 @@
+import React, {useCallback, useMemo, useState, useEffect} from "react"
+import {View, Text, TouchableOpacity, StyleSheet, Linking} from "react-native"
+import * as Clipboard from "expo-clipboard"
+import {
+  useFlowCurrentUser,
+  useCrossVmTokenBalance,
+  UseCrossVmTokenBalanceData,
+  useFlowChainId,
+  getFlowscanAccountUrl,
+  CONTRACT_ADDRESSES,
+} from "@onflow/react-core"
+import type {TokenConfig} from "./Connect"
+import {
+  UserIcon,
+  CopyIcon,
+  CheckIcon,
+  LogOutIcon,
+  ExternalLinkIcon,
+} from "../icons"
+
+type BalanceType = keyof UseCrossVmTokenBalanceData
+
+export interface ProfileProps {
+  /** Callback when user disconnects */
+  onDisconnect?: () => void
+  /** Balance type to display (cadence, evm, or combined) */
+  balanceType?: BalanceType
+  /** Custom tokens to display balance for */
+  balanceTokens?: TokenConfig[]
+}
+
+/**
+ * Profile - Displays user wallet information with balance
+ *
+ * Shows the connected wallet address and token balance.
+ * Provides disconnect functionality and links to Flowscan.
+ *
+ * @example
+ * ```tsx
+ * <Profile
+ *   onDisconnect={() => console.log("Disconnected")}
+ *   balanceType="cadence"
+ * />
+ * ```
+ */
+export const Profile: React.FC<ProfileProps> = ({
+  onDisconnect,
+  balanceType = "cadence",
+  balanceTokens,
+}) => {
+  const {user, unauthenticate} = useFlowCurrentUser()
+  const {data: chainId} = useFlowChainId()
+  const [copied, setCopied] = useState(false)
+
+  // Default token configuration for FlowToken
+  const defaultTokens: TokenConfig[] = useMemo(() => {
+    if (!chainId) return []
+
+    const getFlowTokenAddress = () => {
+      if (chainId === "emulator" || chainId === "local")
+        return CONTRACT_ADDRESSES.local.FlowToken
+      return chainId === "testnet"
+        ? CONTRACT_ADDRESSES.testnet.FlowToken
+        : CONTRACT_ADDRESSES.mainnet.FlowToken
+    }
+
+    const address = getFlowTokenAddress().replace("0x", "")
```

Comment:
We should use the `withPrefix` method we have

Reply:
Yep updated with correct method also on react-sdk profile component for consistency